doc/tut1.rst

```diff
@@ -874,6 +876,31 @@ The example also shows that a proc's body can consist of a single expression
 whose value is then returned implicitly.
 
 
+Funcs and methods
+-----------------
+
+As mentioned in the introduction, Nim differentiates between procedures,
+functions, and methods, defined by the `proc`, `func`, and `method` keywords
+respectively. In some ways, Nim is a bit more pedantic in its definitions than
+other languages.
+
+Functions are closer to the concept of a pure mathematical
+function, which might be familiar to you if you've ever done functional
+programming. Essentially they are procedures with additional limitations set on
+them: they can't access global state (except `const`) and can't produce
+side-effects. The `func` keyword is basically an alias for `proc` tagged
+with `{.noSideEffects.}`. Functions can still change their mutable arguments
+however, which are those marked as `var`, along with any `ref` objects.
```

Comment:
```suggestion
Functions are essentially procedures with additional limitations set on them:
they can't access global state (except `const`) and can't produce side-effects.
This puts Nim functions closer to the concept of a mathematical function, which
might be familiar to you if you've ever done functional programming. The main
benefit of the imposed limits is that it's much easier to reason about the code
with minimized interdependence of its various parts, as data flows directly
through a function from its arguments to its output. The other notable advantage
is that the self-contained nature of a function makes it inherently more
portable and easier to test.

Even though these benefits apply to Nim functions, it's important to keep in
mind that a `func` can still change its mutable arguments: those marked as `var`
along with `ref` objects (which posses interior mutability). This still allows
functions to have additional outputs beside the return value, making them not as
"pure" as their mathematical counterparts.

Technically, the `func` keyword is a shorthand alias for `proc` tagged with
`{.noSideEffects.}`.
```

Reply:
One small typo: it's "possess", not "posses".